connectors/opa/testutil/helper.go

```diff
@@ -0,0 +1,192 @@
+
+	catalogConnectorURL := EnvValues["CATALOG_CONNECTOR_URL"]
+	opaServerUrl := EnvValues["OPA_SERVER_URL"]
+	log.Printf("EnvVariables = %v\n", EnvValues)
+
+	return timeOutSecs, catalogConnectorURL, opaServerUrl
+}
+
+func GetApplicationContext(purpose string) *pb.ApplicationContext {
+	datasetID := "mock-datasetID"
+	applicationDetails := &pb.ApplicationDetails{Purpose: purpose, Role: "Security", ProcessingGeography: "US"}
+	datasets := []*pb.DatasetContext{}
+	datasets = append(datasets, createDatasetRead(datasetID))
+	applicationContext := &pb.ApplicationContext{AppInfo: applicationDetails, Datasets: datasets}
+
+	return applicationContext
+}
+
+func createDatasetRead(datasetID string) *pb.DatasetContext {
+	dataset := &pb.DatasetIdentifier{DatasetId: datasetID}
+	operation := &pb.AccessOperation{Type: pb.AccessOperation_READ}
+	datasetContext := &pb.DatasetContext{Dataset: dataset, Operation: operation}
+	return datasetContext
+}
+
+func ConstructRemoveColumn(colName string) *pb.EnforcementAction {
+	return &pb.EnforcementAction{Name: "removed", Id: "removed-ID",
+		Level: pb.EnforcementAction_COLUMN, Args: map[string]string{"column_name": colName}}
+}
+
+/*func ConstructEncryptColumn(colName string) *pb.EnforcementAction {
+	return &pb.EnforcementAction{Name: "encrypted", Id: "encrypted-ID",
+		Level: pb.EnforcementAction_COLUMN, Args: map[string]string{"column_name": colName}}
+}*/
+
+func ConstructRedactColumn(colName string) *pb.EnforcementAction {
+	return &pb.EnforcementAction{Name: "redact", Id: "redact-ID",
+		Level: pb.EnforcementAction_COLUMN, Args: map[string]string{"column_name": colName}}
+}
+
+/*func ConstructAllow() *pb.EnforcementAction {
+	return &pb.EnforcementAction{Name: "Allow", Id: "Allow-ID",
+		Level: pb.EnforcementAction_DATASET, Args: map[string]string{}}
+}*/
+
+func EnsureDeepEqualDecisions(t *testing.T, testedDecisions *pb.PoliciesDecisions, expectedDecisions *pb.PoliciesDecisions) {
+	assert.True(t, proto.Equal(testedDecisions, expectedDecisions), "Decisions we got from policyManager are not as expected. Expected: %v, Received: %v", expectedDecisions, testedDecisions)
+}
+
+func GetExpectedOpaDecisions(purpose string, in *pb.ApplicationContext) *pb.PoliciesDecisions {
+	var dataset = &pb.DatasetIdentifier{DatasetId: "mock-datasetID"}
+	var datasetDecison *pb.DatasetDecision
+	for _, datasetContext := range in.GetDatasets() {
+		operation := datasetContext.GetOperation()
+		enforcementActions := make([]*pb.EnforcementAction, 0)
+		usedPolicies := make([]*pb.Policy, 0)
+		operation = datasetContext.GetOperation()
+		fmt.Println("operation")
+		fmt.Println(operation)
+		var newUsedPolicy *pb.Policy
+		if purpose == "marketing" {
+			newEnforcementAction := ConstructRedactColumn("nameDest")
+			enforcementActions = append(enforcementActions, newEnforcementAction)
+			newUsedPolicy = &pb.Policy{Description: "reduct columns with name nameOrig and nameDest  in datasets with Finance"}
```

Comment:
finance or marketing?

Reply:
we have been using 2 purposes by convention - fraud-detection and marketing in [other test cases](https://github.com/rohithdv/the-mesh-for-data/blob/6bfa712c3ac2450a9f9eba2d4051384f9141b514/pkg/policy-compiler/pc-bl/policy_compiler_test.go#L62). so we have used "marketing" here since in the opa connector test case context any purpose would do. these are just for testing purposes.

Reply:
This is within a block of an if statement that compares to "marketing"

Reply:
Actually what the description means is “redact columns with name nameOrig and nameDest  in datasets which have been tagged with Finance” ..The purpose could be marketing or fraud-detection or any other purpose..I have edited the description to the above to avoid confusion. Thanks.